api/prisma/createTopics.ts

```diff
@@ -0,0 +1,286 @@
+                else if (topic.newParentCreationId1 && topic.newParentCreationId2) {
+                    parentIds = [created[topic.newParentCreationId1].dbId, created[topic.newParentCreationId2].dbId];
+                }
+                // One of each
+                else if (existingParentId1 && topic.newParentCreationId1) {
+                    parentIds = [existingParentId1, created[topic.newParentCreationId1].dbId];
+                }
+            } else {
+                // Existing parent
+                if (existingParentId1) {
+                    parentIds = [existingParentId1];
+                }
+
+                // New parent
+                if (topic.newParentCreationId1) {
+                    parentIds = [created[topic.newParentCreationId1].dbId];
+                }
+            }
+
+            let createdTopic;
+            const urlBase = `https://${environment === 'int' ? 'int.' : ''}octopus.ac/topics/`;
+
+            if (!dryRun) {
+                try {
+                    createdTopic = await client.prisma.topic.create({
+                        data: {
+                            title: topic.title,
+                            parents: {
+                                connect: parentIds.map((parentId) => ({ id: parentId }))
+                            }
+                        },
+                        select: {
+                            id: true,
+                            parents: {
+                                select: {
+                                    id: true
+                                }
+                            }
+                        }
+                    });
+                } catch (err) {
+                    console.log(err);
+                } finally {
+                    created[topic.creationId] = {
+                        dbId: createdTopic.id,
+                        parentDbIds: parentIds,
+                        title: topic.title,
+                        url: urlBase + String(createdTopic.id)
+                    };
+                }
+            } else {
+                const placeholderDbId = 'localtopic' + topic.creationId;
+                created[topic.creationId] = {
+                    dbId: placeholderDbId,
+                    parentDbIds: parentIds,
+                    title: topic.title,
+                    url: urlBase + placeholderDbId
+                };
+            }
+        }
+    }
+
+    return {
+        created,
+        deferred
+    };
+};
+
+const createTopics = async (
+    data: CreateTopicData[],
+    environment: 'int' | 'prod' = 'int',
+    dryRun?: boolean
+): Promise<CreatedTopics> => {
+    let { created, deferred } = await createOrDeferTopics(data, {}, environment, dryRun);
+    console.log(
+        `First run complete; deferred ${deferred.length} topics and created ${Object.keys(created).length} topics.`
+    );
+    const maxRuns = 5;
+    let runCount = 1;
+
+    while (deferred.length && runCount < maxRuns) {
```

Comment:
What happens to the parent topics you couldn't create if you hit the max runs?

Reply:
I'll add something to report back on which ones couldn't be created